src/elements/toniq-slider/toniq-slider.element.ts

```diff
@@ -0,0 +1,662 @@
+            -webkit-appearance: none;
+            position: relative;
+            height: 16px;
+            width: 16px;
+            z-index: 10;
+            cursor: pointer;
+            pointer-events: all;
+            border-radius: 10px;
+            transition: ${interactionDuration};
+            ${applyBackgroundAndForeground(toniqColors.accentPrimary)};
+        }
+
+        .slider::-webkit-slider-thumb:hover {
+            transform: scale(1.2);
+        }
+    `,
+    renderCallback: ({props, host, events, dispatch, setProps}) => {
+        const progress = props.referenceElements?.progress;
+        const slider = props.referenceElements?.slider;
+        const label = props.referenceElements?.label;
+        const lowerSliderElement = props.referenceElements?.lowerSlider;
+        const upperSliderElement = props.referenceElements?.upperSlider;
+        const lowerLabelElement = props.referenceElements?.lowerLabel;
+        const upperLabelElement = props.referenceElements?.upperLabel;
+        const doubleSliderOffset = 2;
+
+        if (props.double) {
+            if (!isDoubleRangeValue(props.value)) {
+                setProps({
+                    value: {
+                        min: props.min,
+                        max: props.max,
+                    },
+                });
+            } else if (
+                lowerSliderElement instanceof HTMLInputElement &&
+                upperSliderElement instanceof HTMLInputElement
+            ) {
+                const defaultMin = 0;
+                const defaultMax = 100;
+                const initValue = {
+                    min:
+                        props.value.min < props.min
+                            ? props.min
+                            : props.value.min > props.max
+                            ? props.max
+                            : parseInt(lowerSliderElement.value) !== defaultMin
+                            ? parseInt(lowerSliderElement.value)
+                            : props.value.min,
+                    max:
+                        props.value.max < props.min
+                            ? props.min
+                            : props.value.max > props.max
+                            ? props.max
+                            : parseInt(upperSliderElement.value) !== defaultMax
+                            ? parseInt(upperSliderElement.value)
+                            : props.value.max,
+                };
+                setProps({
+                    value: initValue,
+                    lowerLabelStyle: {
+                        innerText: `${initValue.min} ${props.suffix}`,
+                    },
+                    upperLabelStyle: {
+                        innerText: `${initValue.max} ${props.suffix}`,
+                    },
+                });
+            }
+
+            fillDoubleRangeSlider();
+        } else {
+            const initValue =
+                props.value < props.min
```

Comment:
This can be accomplished like so:
```TypeScript
Math.min(Math.max(props.value, props.min), props.max)
```

That could also be put in a `clamp` function inside of `augments/number.ts` for reuse :D